respy/pre_processing/model_processing.py

```diff
@@ -144,175 +92,62 @@ def parse_parameters(paras_vec, paras_type="optim"):
         aligned with Fortran, where we never have to parse 'econ' parameters.
 
     """
-    paras_vec = paras_vec.copy()
 
-    pinfo = _paras_parsing_information(len(paras_vec))
-    optim_paras = {}
+    if isinstance(params, pd.DataFrame):
+        params = params["para"]
+    elif isinstance(params, pd.Series):
+        pass
+    else:
+        raise ValueError("Invalid params in parse_parameters: {}.".format(type(params)))
```

Comment:
TypeError and maybe different message: Invalid type {} for params. And maybe f-strings.